galaxy/tools/deps/mulled/mulled_build.py

```diff
@@ -0,0 +1,274 @@
+#!/usr/bin/env python
+"""Creates shell commands to run involucro.
+
+Examples
+
+Build relevant images with:
+
+    ./helpers.py build
+
+Build, test, and publish relevant images with:
+
+    ./helpers.py all
+
+"""
+from __future__ import print_function
+
+import argparse
+import collections
+import json
+import os
+import re
+import string
+import subprocess
+import sys
+import time
+
+from galaxy.tools.deps import commands
+
+try:
+    import yaml
+except ImportError:
+    yaml = None
+
+try:
+    from conda_build.metadata import MetaData
+except ImportError:
+
+    class MetaData(object):
+
+        def __init__(self, input_dir):
+            recipe_path = os.path.join(input_dir, "meta.yaml")
+            self.input_dir = input_dir
+            with open(recipe_path, "r") as f:
+                self.meta = yaml.load(f)
+
+        def get_value(self, field, default=None):
+            """Get nested field value or supplied default is not present."""
+            section, key = field.split('/')
+            submeta = self.meta.get(section)
+            if submeta is None:
+                submeta = {}
+            res = submeta.get(key)
+            if res is None:
+                res = default
+            return res
+
+from .util import quay_versions, split_tag
+
+DIRNAME = os.path.dirname(__file__)
+
+
+def natural_key(string_):
+    """See http://www.codinghorror.com/blog/archives/001018.html."""
+    return [int(s) if s.isdigit() else s for s in re.split(r'(\d+)', string_)]
+
+
+def get_tests(args, pkg_path):
+    """Extract test cases given a recipe's meta.yaml file."""
+    recipes_dir = args.recipes_dir
+
+    tests = ""
+    input_dir = os.path.dirname(os.path.join(recipes_dir, pkg_path))
+    recipe_meta = MetaData(input_dir)
+
+    tests_commands = recipe_meta.get_value('test/commands')
+    tests_imports = recipe_meta.get_value('test/imports')
+    requirements = recipe_meta.get_value('requirements/run')
+
+    if tests_imports or tests_commands:
+        if tests_commands:
+            tests = ' && '.join(tests_commands)
```

Comment:
Poor mans testing, I feel a little bit ashamed. I need to have a better idea here.